api/v1alpha1/trafficmanagerprofile_types.go

```diff
@@ -0,0 +1,143 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TrafficManagerProfile struct {
```

Comment:
in my understanding, the class means it would have different implementations. For the traffic manager, i don't think we'll have multiple implementations. it's very unique to Azure.